### PR TITLE
feat: remove lightning

### PR DESCRIPTION
**Pull request by stephantul**
This PR removes all references to lightning and pytorch lightning. Scores are (on average) identical for classification, but the new training loop is faster. Since the core training loop is shared by all training variants, I have high confidence that this also works for multi-label and regression tasks.

### CODE REVIEW

File: model2vec/train/trainer.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    model.eval()
+    accumulated: dict[str, list[float]] = defaultdict(list)
+    for x, y in val_loader:
+        x, y = x.to(device), y.to(device)
+        head_out = model(x)
+        loss = loss_function(head_out, y)
+        for key, value in compute_metrics(head_out, y, loss).items():
+            accumulated[key].append(value)
+    model.train()
+    return {key: sum(values) / len(values) for key, values in accumulated.items()}
```
Comment (Pringled) [inline model2vec/train/trainer.py]: Don't we need to weight this by batchsize? Otherwise the samples from the last batch (which is usually smaller) contribute more than they should.

File: model2vec/train/base.py [hunk 2]
```diff
@@ -222,10 +218,9 @@ def encode(self, X: list[str], batch_size: int = 1024, show_progress_bar: bool =
 
         return np.concatenate(pred, axis=0)
 
-    def forward(self, input_ids: torch.Tensor) -> tuple[torch.Tensor, torch.Tensor]:
+    def forward(self, input_ids: torch.Tensor) -> torch.Tensor:
```
Comment (Pringled) [inline model2vec/train/base.py:221]: Technically a breaking API change but not sure if it matters, wdyt? We could also just keep this as is and have the trainer ignore the extra value, but then again I'm not sure if anyone calls forward directly.

**Review by Pringled — APPROVED** [review]
Very nice! Two small comments but looks good to me

File: model2vec/train/trainer.py (hunk 1 above)
Comment (stephantul) [reply model2vec/train/trainer.py]: Good point
